tests/components/broadlink/test_config_flow.py

```diff
@@ -819,3 +820,143 @@ async def test_flow_reauth_valid_host(hass):
     assert mock_entry.data["host"] == device.host
     assert mock_discover.call_count == 1
     assert mock_api.auth.call_count == 1
+
+
+async def test_dhcp_can_finish(hass):
+    """Test DHCP discovery flow can finish right away."""
+    await setup.async_setup_component(hass, "persistent_notification", {})
+
+    device = get_device("Living Room")
+    device.host = "1.2.3.4"
+    mock_api = device.get_mock_api()
+
+    with patch(DEVICE_DISCOVERY, return_value=[mock_api]):
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN,
+            context={"source": "dhcp"},
+            data={
+                HOSTNAME: "broadlink",
+                IP_ADDRESS: "1.2.3.4",
+                MAC_ADDRESS: "aa:bb:cc:dd:ee:ff",
```

Comment:
The mock comes with host, mac, type etc. We should abstract this information so we can easily replace the devices in the tests:
```python3
                HOSTNAME: "broadlink",
                IP_ADDRESS: device.host[0],
                MAC_ADDRESS: device.mac,
```

Reply:
Ah, just realized you need the MAC in a different format. That's ok.

Reply:
Sorry saw your comment after I merged.  I'll adjust that in a new pr shortly

Reply:
Fixed here https://github.com/home-assistant/core/pull/48415